packages/cli-hydrogen/package.json

```diff
@@ -47,13 +47,16 @@
     "@shopify/cli-kit": "3.25.0",
     "fast-glob": "3.2.11",
     "fs-extra": "10.0.0",
-    "typescript": "4.8.4",
+    "typescript": "4.6.4",
```

Comment:
To be consistent with the other `package.json`s in the workspace. We should be using the same version across all the projects to prevent inconsistent behaviours.